Makefile

```diff
@@ -0,0 +1,72 @@
+	@echo 'local development of jenkins jobs and features, but may also facilitate with'
+	@echo 'production deploys for the OpenEdX community.'
+	@echo ''
+	@echo 'Available targets:'
+	@echo '    help                       show this information'
+	@echo '    clean                      shutdown and delete running containers'
+	@echo '    docker.run.$$service        run the specified jenkins service container'
+	@echo '    docker.running.$$service    check if specified jenkins service container is running'
+	@echo '    docker.configure.$$service  configure the specified jenkins service container'
+	@echo '    docker.seed_job.$$service   seed a job specified by the following two environment variables: DSL_SCRIPT, BRANCH'
+
+clean: docker.clean
+
+docker.clean:
+	docker-compose down
+
+# docker.run.$service
+# This target fetches the docker image corresponding to the requested jenkins
+# service and launches it into a new container.  If this is used to start
+# jenkins again after stopping it, jenkins jobs and configuration will persist.
+$(DOCKER_SERVICES:%=docker.run.%) : docker.run.% :
+	docker-compose up -d $*
+
+# docker.running.$service
+# Check if service is running.
+$(DOCKER_SERVICES:%=docker.running.%) : docker.running.% :
+	docker-compose exec $* /bin/bash -c 'echo "'$*' is running"'
+
+# docker.configure.$service
+# This target picks up any user-specified ansible overrides, then reconfigures
+# and restarts the specified service container.
+docker.configure.jenkins_build : docker.configure.% : ansible_overrides.yml docker.running.%
+	docker cp ansible_overrides.yml $*:/ansible_overrides_extra.yml
+	# The jenkins:local-dev ansible tag is specific to jenkinses that use
+	# the jenkins_common role.
+	docker-compose exec $* /bin/bash -c "PYTHONUNBUFFERED=1 /edx/app/edx_ansible/venvs/edx_ansible/bin/ansible-playbook \
+		-v $*.yml \
+		-i '127.0.0.1,' \
+		-c local \
+		-e@/ansible_overrides.yml \
+		-e@/ansible_overrides_extra.yml \
+		-t 'jenkins:local-dev' \
+		-vv"
+	docker-compose restart $*
+
+# Reconfigure jenkins_tools
+docker.configure.jenkins_tools:
+	@echo 'This makefile currently does not support reconfiguring jenkins_tools.'
+	exit 1
+
+# docker.install_cli.$service
+# "/jnlpJars/jenkins-cli.jar" is the standard location for the jar.
+# Every jenkins master has this file.
+# More info: https://jenkins.io/doc/book/managing/cli/#using-the-cli-client
+$(DOCKER_SERVICES:%=docker.install_cli.%) : docker.install_cli.% :
+	docker-compose exec $* /bin/bash -c 'curl --output /opt/jenkins-cli.jar http://localhost:8080/jnlpJars/jenkins-cli.jar'
+
+# docker.seed_job.$service
+$(DOCKER_SERVICES:%=docker.seed_job.%) : docker.seed_job.% :
+	docker-compose exec $* /bin/bash -c 'su -l jenkins --shell=/bin/bash -c "java -jar /opt/jenkins-cli.jar -auth admin:$$(cat /var/lib/jenkins/secrets/initialAdminPassword) build -v -s -p DSL_SCRIPT='$(DSL_SCRIPT)' manually_seed_one_job"'
```

Comment:
@estute can you imagine a more succinct and robust way of writing this?  Specifically, it does not handle the case where the administrator changes the admin password from the default.  I looked over all the CLI options, but it looks like one way or another the local client needs to authenticate, even though we have direct access to the filesystem.